llvm/include/llvm/CodeGen/TargetInstrInfo.h

```diff
@@ -136,6 +136,10 @@ class TargetInstrInfo : public MCInstrInfo {
                                          const TargetRegisterInfo *TRI,
                                          const MachineFunction &MF) const;
 
+  /// Returns true if MI is an instruction we are unable to reason about
+  /// (like a call or something with unmodeled side effects).
+  virtual bool isGlobalMemoryObject(const MachineInstr *MI) const;
```

Comment:
This should follow the pattern that isCopyInstr uses. Have the virtual overridable Impl function, with the generic cases handled directly